### DISCUSSION TITLE
Nested span trace id not inherited from parent's active span

### DISCUSSION THREAD

**Discussion by dluongiop** [open-telemetry/opentelemetry-cpp#2211] [Q&A] [answered]
Hi.

Referring to https://opentelemetry-cpp.readthedocs.io/en/latest/api/GettingStarted.html#create-nested-spans

According to documentation, the nested span's trace id is supposed to be inherited from the active span but when I tried to construct the nested span as in the documentation, it is not unless I pass the StartSpanOptions with the parent's context to StartSpan() the child span. Is the child span supposed to get the active span's trace id without passing options or did I miss something?

Please advise.
Thank you.

**Comment by lalitb** [comment]
> Is the child span supposed to get the active span's trace id without passing options or did I miss something?

Yes child span should inherit the trace id from parent. You can go through the nested span example here - https://github.com/open-telemetry/opentelemetry-cpp/blob/main/examples/simple/main.cc ( refer - call to `foo_library()` function)
If it still doesn't work. please share your code here.

**Reply by dluongiop** [reply]
I tried doing exactly as foo_library() in the example but it didn't inherit the trace id.  Below is a version using WithactiveSpan() call with the same result.

```
void Example::Run()
{
    f2();
}

void Example::f1()
{
    auto span = GetTracer()->StartSpan("f1");
    auto scope = GetTracer()->WithActiveSpan(span);
    span->End();
}

void Example::f2()
{
    auto span = GetTracer()->StartSpan("f2");
    auto scope = GetTracer()->WithActiveSpan(span);
    f1();
    span->End();
}
```

output:
```
{
  name          : f1
  trace_id      : 6883353f0e6cb3bb9f3756e464a1ba82
  span_id       : 9cac9ced165a9144
  tracestate    :
  parent_span_id: 0000000000000000
  start         : 1688075439796258634
  duration      : 1059
  description   :
  span kind     : Internal
  status        : Unset
  attributes    :
  events        :
  links         :
  resources     :
    service.name: unknown_service
    telemetry.sdk.version: 1.8.3
    telemetry.sdk.language: cpp
    telemetry.sdk.name: opentelemetry
  instr-lib     : optl_example-1.0.0
}
{
  name          : f2
  trace_id      : 4384afbfb944b333497db7afeb8de512
  span_id       : 827b82337e84bb67
  tracestate    :
  parent_span_id: 0000000000000000
  start         : 1688075439796249688
  duration      : 872775
  description   :
  span kind     : Internal
  status        : Unset
  attributes    :
  events        :
  links         :
  resources     :
    service.name: unknown_service
    telemetry.sdk.version: 1.8.3
    telemetry.sdk.language: cpp
    telemetry.sdk.name: opentelemetry
  instr-lib     : optl_example-1.0.0
}
```

**Reply by lalitb** [reply]
Which platform, compiler and it's version ?

**Reply by dluongiop** [reply]
Amazon Linux2
C, CXX compiler is GNU 7.3.1

I built the otel SDK from source because I couldn't find any prebuilt packages that can be installed on AWS EC2 instance.

**Reply by lalitb** [reply, marked as answer]
Can you please check and confirm if this patch fixes the issue - https://github.com/open-telemetry/opentelemetry-cpp/issues/1014#issuecomment-944821770.

**Reply by dluongiop** [reply]
The patch did fix the issue.
